sdk/core/azure-core/src/main/java/com/azure/core/util/ETag.java

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util;
+
+import com.azure.core.util.logging.ClientLogger;
+
+/**
+ * This class represents an HTTP ETag.
+ */
+public final class ETag {
+    private static final ClientLogger LOGGER = new ClientLogger(ETag.class);
+
+    private static final String EMPTY_STRING = "";
+    private static final String QUOTE_STRING = "\"";
+    private static final String WEAK_ETAG_PREFIX_QUOTE = "W/\"";
+
+    public static final String DEFAULT_FORMAT = "G";
+    public static final String HEADER_FORMAT = "H";
```

Comment:
I know this was copied from .NET's implementation, but are these values specific to the .NET `string.format` method? If so, do we need these?